File: algorithms/S/Sort Array By Parity/Sort Array By Parity.java
class Solution {
        
    public int[] sortArrayByParity(int[] nums) {
        int i = 0;
        int j = 0;
        
        while(i < nums.length){
            if(nums[i] % 2 == 1){
                i++;
            }else{
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                
                i++;
                j++;
            }
        }
        
        return nums;       
    }
}

File: algorithms/S/Sort Array By Parity/Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int n = nums.size();
        if (n == 1)
        {
            return nums;
        }
        
        int j = n - 1;
        for (int i = 0; i < j; i ++)
        {
            while(j > i)
            {
                if (nums[j] % 2 == 0)
                {
                    break;
                }
                j --;
            }
            
            if (nums[i] % 2)
            {
                swap(nums[i], nums[j]);
            }
        }
        
        return nums;
    }
};
